File: premierprint/public/js/sales_order.js
frappe.ui.form.on('Sales Order', {
    onload: function (frm) {
        // Yangi hujjat ochilganda standart seriyani bo'shatib turish yoki kompaniyaga qarab o'rnatish
        if (frm.is_new() && frm.doc.company) {
            apply_naming_series(frm);
        }
    },
    company: function (frm) {
        apply_naming_series(frm);
    }
});

function apply_naming_series(frm) {
    const mapping = {
        "Premier Print": "ПП-.#######",
        "Полиграфия": "П-.########",
        "Реклама": "Р-.#######",
        "Сувенир": "С-.#######"
    };

    if (frm.doc.company && mapping[frm.doc.company]) {
        console.log("Applying Naming Series for:", frm.doc.company, "->", mapping[frm.doc.company]);
        frm.set_value('naming_series', mapping[frm.doc.company]);
        // frappe.msgprint msg removed as per user request to avoid distraction
    }
}

File: premierprint/utils/sales_order.py
import frappe

def set_naming_series(doc, method):
    mapping = {
        "Premier Print": "ПП-.#######",
        "Полиграфия": "П-.########",
        "Реклама": "Р-.#######",
        "Сувенир": "С-.#######"
    }

    if doc.company in mapping:
        # Tizimga aniq formatni (panjaralari bilan) majburlab ko'rsatamiz
        doc.naming_series = mapping[doc.company]